File: src/homemade/game/SelectionState.java
package homemade.game;

import homemade.game.fieldstructure.CellCode;

/**
 * Created by user3 on 25.03.2016.
 */
public interface SelectionState
{
    public boolean isSelected(CellCode cellCode);
    public boolean canMoveTo(CellCode cellCode);
}

File: game-model-v2/src/homemade/game/state/SelectionState.kt
package homemade.game.state

import homemade.game.fieldstructure.CellCode

interface SelectionState {
    val selection: CellCode?
    val cellsToMove: Set<CellCode>
}

fun SelectionState.isSelected(cellCode: CellCode): Boolean {
    return selection == cellCode
}

fun SelectionState.canMoveTo(cellCode: CellCode): Boolean {
    return cellCode in cellsToMove
}
